src/plugins/socketmon/socketmon.cpp

```diff
@@ -1379,6 +1382,287 @@ static event_response_t udpb_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+    unicode_string_t* domain_name_us = nullptr;
+
+    if (sm->pm == VMI_PM_IA32E) // only 64 bit for now
+    {
+        domain_name_addr = info->regs->rcx;
+
+        access_context_t ctx;
+        ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+        ctx.dtb = info->regs->cr3;
+        ctx.addr = domain_name_addr;
+
+        {
+            vmi_lock_guard vmi_lg(drakvuf);
+            domain_name_us = drakvuf_read_wchar_string(vmi_lg.vmi, &ctx);
+        }
+
+        if (domain_name_us)
+        {
+            print_dns_info(drakvuf, info, sm, info->trap->name, (char*)domain_name_us->contents);
+        }
+        else
+        {
+            fprintf(stderr, "[SOCKETMON] Error, getting unicode domain name string in %s()", __FUNCTION__);
+        }
+        vmi_free_unicode_str(domain_name_us);
+    }
+    else
+    {
+        fprintf(stderr, "[SOCKETMON] Suddenly, we are in 32 bit mode in %s(...) trap. Unsupported.\n", info->trap->name);
+    }
+
+
+    return 0;
+}
+
+// Works on Windows 7
+static event_response_t trap_DnsQueryExW_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    socketmon* sm = (socketmon*)info->trap->data;
+
+    if (sm->pm != VMI_PM_IA32E) // only 64 bit for now
+    {
+        fprintf(stderr, "[SOCKETMON] Suddenly, we are in 32 bit mode in %s(...) trap. Unsupported.\n", info->trap->name);
+        return 0;
+    }
+
+    unicode_string_t* domain_name_us = drakvuf_read_unicode(drakvuf, info, info->regs->rcx);
+
+    if (domain_name_us)
+    {
+        print_dns_info(drakvuf, info, sm, info->trap->name, (char*)domain_name_us->contents);
+    }
+    else
+    {
+        fprintf(stderr, "[SOCKETMON] Error, getting unicode domain name string in %s()", __FUNCTION__);
+    }
+
+    vmi_free_unicode_str(domain_name_us);
+
+    return 0;
+}
+
+// Works on Windows 7
+static event_response_t trap_DnsQueryExA_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    socketmon* sm = (socketmon*)info->trap->data;
+
+    addr_t domain_name_addr = 0;
+
+    if (sm->pm == VMI_PM_IA32E) // only 64 bit for now
+    {
+        domain_name_addr = info->regs->rcx;
+
+        access_context_t ctx;
+        ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+        ctx.dtb = info->regs->cr3;
+        ctx.addr = domain_name_addr;
+
+        char* dns_name = [&]
+        {
+            vmi_lock_guard vmi_lg(drakvuf);
+            return vmi_read_str(vmi_lg.vmi, &ctx);
+        }();
+
+        print_dns_info(drakvuf, info, sm, info->trap->name, dns_name);
+        g_free(dns_name);
+    }
+    else
+    {
+        fprintf(stderr, "[SOCKETMON] Suddenly, we are in 32 bit mode in %s(...) trap. Unsupported.\n", info->trap->name);
+    }
+
+    return 0;
+}
+
+// Works on Windows 8+
+static event_response_t trap_DnsQueryEx_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    socketmon* sm = (socketmon*)info->trap->data;
+
+    unicode_string_t* domain_name_us = nullptr;
+
+    if (sm->pm == VMI_PM_IA32E) // only 64 bit for now
```

Comment:
These traps shouldn't even be registered on 32-bit systems.

Reply:
Good idea!

Reply:
Added this check before registering traps.